### PR TITLE
GEODE-4133: synchronize `ClientHealthMonitor.getClientHeartbeats`

### PR DESCRIPTION
**Pull request by galen-pivotal**
Add generics and make some things private while I'm at it. eventually maybe we can fix GEODE-2678.

- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [I don't think this is necessary, be] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

@WireBaron @PivotalSarge

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java [hunk 1]
```diff
@@ -59,18 +59,20 @@
...
+   * Accesses must be locked by _clientHeartbeatsLock
    */
-  protected volatile Map _clientHeartbeats = Collections.EMPTY_MAP;
+  private Map<ClientProxyMembershipID, Long> _clientHeartbeats = Collections.emptyMap();
 
   /**
    * An object used to lock the map of known clients
    */
-  protected final Object _clientHeartbeatsLock = new Object();
+  private final Object _clientHeartbeatsLock = new Object();
```
Comment (PivotalSarge) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java]: Perhaps remove the leading underscores on variables? Hungarian notation is the root of all evil.

**Comment by galen-pivotal** [comment]
@upthewaterspout It does look like copy-on-write. It was giving access to the actual map in `getClientHeartbeats` that really stuck out to me; I was just trying to do some cleanup with the rest of it. It may be fine to wrap the returned map in an `UnmodifiableMap`.

I'm somewhat reluctant to use an actual `CopyOnWriteHashMap` because that would change semantics, the code isn't all that well tested, and I wanted to do a quick improvement and make the heartbeats map not publically available.

**Comment by upthewaterspout** [comment]
@galen-pivotal . Ok, makes sense. Looks good then.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java (hunk 1 above)
Comment (galen-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitor.java]: I could, but it's consistent with the other fields on this class, and I'm trying to keep this a quick, minimally invasive change.

**Pull request merged**